crates/trie/trie/src/node_iter.rs

```diff
@@ -156,6 +159,9 @@ where
     /// If `metrics` feature is enabled, also updates the metrics.
     fn next_hashed_entry(&mut self) -> Result<Option<(B256, H::Value)>, DatabaseError> {
         let result = self.hashed_cursor.next();
+
+        self.last_next_result = result.clone()?;
+        self.last_seeked_hashed_entry = None;
```

Comment:
same, no need to reset it, as we can re-use it later

Reply:
Done